db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java

```diff
@@ -82,23 +135,166 @@ public static String toString(PredicateList predicateList) {
     /**
      * Return string representation of a Derby expression
      */
-    public static String opToString(ValueNode operand){
+    public static String opToString(ValueNode operand) {
+        SPARK_EXPRESSION.set(Boolean.FALSE);
+        initRelationalOpDepth();
+        try {
+            return opToString2(operand);
+        }
+        catch(StandardException e) {
+            return "Bad SQL Expression";
+        }
+
+    }
+
+    /**
+     * Return a spark SQL expression given a Derby SQL expression, with column
+     * references indicating column names in the source DataFrame.
+     */
+    public static String opToSparkString(ValueNode operand) {
+        String retval = null;
+
+        // Do not throw any errors encountered.  An error condition
+        // just means we can't generate a valid spark representation
+        // of the SQL expression to apply to a native spark Dataset,
+        // so should not be considered a fatal error.
+        try {
+            SPARK_EXPRESSION.set(Boolean.TRUE);
```

Comment:
We should pass down a parameter into opToString2() and related functions rather than using a ThreadLocal. If we are not sharing much code in opToString2() please consider using a completely separate function for generating the Spark representation, in most cases it seems we are not sharing anything at all.

Reply:
> We should pass down a parameter into opToString2() and related functions rather than using a ThreadLocal. If we are not sharing much code in opToString2() please consider using a completely separate function for generating the Spark representation, in most cases it seems we are not sharing anything at all.

Moved the thread local variables into a "vars" parameter, collectively held together in an instantiated OperatorToString object, to make the code more legible (some lines of code call OperatorToString multiple times).  This also makes it easier to add new parameters in the future.

I considered using a separate function originally, but thought that some of the changes I'm adding might be good to also incorporate into explain output at some point.  Currently some SQL expressions aren't fully printed in the explain output, so we are not getting full information.